kotlinx-coroutines-core/common/test/TestBase.common.kt

```diff
@@ -43,6 +49,16 @@ public expect open class TestBase constructor() {
     ): TestResult
 }
 
+public fun TestBase.runMtTest(
+    expected: ((Throwable) -> Boolean)? = null,
+    unhandled: List<(Throwable) -> Boolean> = emptyList(),
+    block: suspend CoroutineScope.() -> Unit
+): TestResult {
+    @Suppress("CAST_NEVER_SUCCEEDS")
+    if (!multithreadingSupported) return Unit as TestResult
```

Comment:
Added

```
    // Never invoked on JS, not extracted to separate 'concurrent' source set
    // as it doesn't help with the suppress anyway and requires an additional ceremony
    // and spreads TestBase extensions over the codebase
```

Reply:
I see. I do think that this ceremony would be warranted, given that it would at least prevent `runMtTest` autocompletion from popping up anywhere other than inside `concurrent`. On the other hand, we may hope that `runMtTest` will only exist for a short while anyway, so it shouldn't be a huge annoyance.

Reply:
I've addressed your concern along with `multithreadingSupport` below